src/pip/_internal/wheel.py

```diff
@@ -881,11 +886,12 @@ def _build_one_legacy(self, req, tempd, python_tag=None):
             try:
                 call_subprocess(wheel_args, cwd=req.setup_py_dir,
                                 show_stdout=False, spinner=spinner)
-                return True
             except Exception:
                 spinner.finish("error")
                 logger.error('Failed building wheel for %s', req.name)
-                return False
+                return None
+            else:
+                return os.listdir(tempd)[0]
```

Comment:
> It's the wheel name in both cases.

Okay, it looks like you're passing `os.path.join(tempd, newname)` in the other method, so that would need to be changed then.

Reply:
Yea, nice catch! I'll probably make them both full paths.
(on mobile, apologies for terseness)

Reply:
Full paths, I think that's clearer, too.